File: dto-mapping-benchmark/src/main/java/com/example/dtomappingbenchmark/v4/mapper/PostMapper.java
package com.example.dtomappingbenchmark.v4.mapper;

import com.example.dtomappingbenchmark.v4.entity.Post;
import com.example.dtomappingbenchmark.v4.entity.User;
import com.example.dtomappingbenchmark.v4.dto.PostDto;

public class PostMapper {
    public static Post toEntity(PostDto dto){
        return Post.builder()
                .id(dto.getId())
                .title(dto.getTitle())
                .content(dto.getContent())
                .user(null)
                .build();
    }

    public static void patchUser(User user, Post post){
        post.setUser(user);
    }

    public static PostDto toDto(Post entity){
        return PostDto.builder()
                .id(entity.getId())
                .title(entity.getTitle())
                .content(entity.getContent())
                .userId(entity.getUser().getId())
                .build();
    }
}

File: object-mapping-benchmark-v4/src/main/kotlin/yu/selab/cj/objectmappingbenchmarkv4/mapper/PostMapper.kt
package yu.selab.cj.objectmappingbenchmarkv4.mapper

import yu.selab.cj.objectmappingbenchmarkv4.post.command.CreatePostDto
import yu.selab.cj.objectmappingbenchmarkv4.post.command.ReadPostDto
import yu.selab.cj.objectmappingbenchmarkv4.post.entity.Post
import yu.selab.cj.objectmappingbenchmarkv4.user.entity.User


object PostMapper {

    // 새로운 게시물 쓰기
    fun toEntity(dto : CreatePostDto, user : User) : Post {
        return Post(
            title = dto.title,
            content = dto.content,
            user = user
        )
    }

    fun toDto(post : Post) : ReadPostDto {
        return ReadPostDto(
            post.pid,
            post.title,
            post.content
        )
    }

}